src/modules/0.0.22/aws_s3/mappers/bucket.ts

```diff
@@ -135,7 +129,8 @@ export class BucketMapper extends MapperBase<Bucket> {
         const { bucketName, region } = this.idFields(id);
         if (enabledRegions.includes(region)) {
           const allBuckets = await this.getBuckets(client.s3Client);
-          rawBuckets = allBuckets.filter(b => !bucketName || b.Name === bucketName).filter(b => !!b.Name);
+          const foundBucket = allBuckets.find((b: BucketAWS) => b.Name === bucketName);
+          if (foundBucket) rawBuckets.push(foundBucket);
```

Comment:
this is not solving my previous comment. This way you would still be returning a list of buckets instead of a single bucket when the `cloud.read` method is called asking just for one `id`, it is not consistent.